### PR TITLE
Update order books and matching engine

### PR DESCRIPTION
**Pull request by aaronbuchwald**

### CODE REVIEW

**Review by jackrehmann — APPROVED** [review]

File: matchingEngine.ml [hunk 1]
```diff
@@ -1,21 +1,152 @@
...
+     | Some order -> begin
+        (* let (addr, o_amt, price, time) = order in 
+           let tx_amount = min o_amt amount in 
+           let tx = price, tx_amount, addr  *)
+        ob'
+      end
+     | None -> ob'
+
+     let execute_market_buy (me: AccountManager.t) (ob: OrderBook.t) (amount: int) : OrderBook.t = 
+     failwith "" *)
```
Comment (seancorc) [inline matchingEngine.ml]: Is this commented out section needed?

File: main.ml [hunk 2]
```diff
@@ -104,57 +90,94 @@ let login (s : State.t) : State.t =
...
-let restart (s : State.t) : State.t  = 
+let restart (s : state) : state  = 
   match start_loop () with 
   | Login -> login s
   | Signup -> register s
+(* 
+let parse_details usr amount price = 
+let amount' = int_of_string amount in 
+let price' = float_of_string in 
+(usr, ) *)
```
Comment (seancorc) [inline main.ml]: Is this commented out section needed?

File: main.ml [hunk 3]
```diff
@@ -23,32 +24,32 @@ let startup_action () =
...
-  | [] -> ()
-  | (o1,o2)::t -> 
+   let parse_txns (txns: transaction list) am = 
+   match txns with 
+   | [] -> ()
+   | (o1,o2)::t -> 
     let _ = execute_order am o1 in 
     let _ = execute_order am o2 in 
-    ()
+    () *)
```
Comment (seancorc) [inline main.ml]: Is there a reason for all this commented out stuff?

File: accountManager.mli [hunk 4]
```diff
@@ -12,45 +12,57 @@ exception InvalidPassword
...
+  val register : t -> string -> string -> Account.t
+  val set_account_balance : t -> string -> float -> unit
+  val inc_account_balance : t -> string -> float -> unit
+  val dec_account_balance : t -> string -> float -> unit
+  val set_account_position : t -> string -> string -> int -> unit
+  val inc_account_position : t -> string -> string -> int -> unit
+  val dec_account_position : t -> string -> string -> int -> unit
+  val login : t -> string -> string -> Account.t
+  val delete_user : t -> string -> string -> unit
   val accounts : t -> string list
```
Comment (seancorc) [inline accountManager.mli]: Why did you change to this format for the mli?

File: orderBook.ml [hunk 5]
```diff
@@ -33,57 +49,157 @@ module OrderBook : OrderBook = struct
...
+     List.filter (fun elt -> not (elt = o)) lst
 
-  let remove (o : order) ((b, s) : t) = 
-    match o with 
-    | {asset; price; order_type = Buy; username} -> (remove_helper o b, s)
-    | {asset; price; order_type = Sell; username} -> (b, remove_helper o s)
+     let remove (o : order) ((b, s) : t) = 
+     match o with 
+     | {asset; price; order_type = Buy; username} -> (remove_helper o b, s)
+     | {asset; price; order_type = Sell; username} -> (b, remove_helper o s) *)
```
Comment (seancorc) [inline orderBook.ml]: Same question with keeping commented out code.

File: orderBook.mli [hunk 6]
```diff
@@ -1,50 +1,89 @@
...
-(** [OrderBook] is the signature for the module type that will hold all of the 
+   type order = {
+   asset: string;
+   price: float;
+   order_type: order_type;
+   username: string
+   }
+
+   (** [OrderBook] is the signature for the module type that will hold all of the 
     orders in the market. *)
```
Comment (seancorc) [inline orderBook.mli]: same question with commented out code

**Review by seancorc — APPROVED** [review]
Delete the commented out section in matchingEngine.ml before squashing and merging
